flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/io/benchmark/StreamNetworkBenchmarkEnvironment.java

```diff
@@ -272,11 +287,161 @@ private static ShuffleDescriptor createShuffleDescriptor(
 			ResultPartitionID resultPartitionID,
 			ResourceID location,
 			TaskManagerLocation senderLocation,
-			int channel) {
+			int connectionIndex) {
 		final NettyShuffleDescriptorBuilder builder = NettyShuffleDescriptorBuilder.newBuilder()
 			.setId(resultPartitionID)
 			.setProducerInfoFromTaskManagerLocation(senderLocation)
-			.setConnectionIndex(channel);
+			.setConnectionIndex(connectionIndex);
 		return localMode ? builder.setProducerLocation(location).buildLocal() : builder.buildRemote();
 	}
+
+	/**
+	 * A {@link SingleInputGateFactory} which replaces the default {@link RemoteInputChannel} and
+	 * {@link LocalInputChannel} implementation with costume ones.
+	 */
+	private static class TestSingleInputGateFactory extends SingleInputGateFactory {
```

Comment:
nit: We can make it a separate class to void this class too long. Also rename it `SingleInputGateBenchmarkFactory`.